libyul/AsmAnalysis.cpp

```diff
@@ -52,7 +52,8 @@ bool AsmAnalyzer::analyze(Block const& _block)
 	if (!(ScopeFiller(m_info, m_errorReporter))(_block))
 		return false;
 
-	return (*this)(_block);
+	bool success = (*this)(_block);
+	return success && !m_errorReporter.hasErrors();
```

Comment:
Is it guaranteed that `exception -> m_errorReporter.hasErrors()`?

Reply:
What do you mean by `execption`?

The problem was that this component found an error but due to the internal logic, it was difficult to return `false` at that point. I changed it to just also include any errors found but not explicitly flagged by return.

Reply:
Separately, do we need an assertion for `!success && hasErrors()` ?

Reply:
no, this fixes a bug where we returned success but there were errors, so the assertion will trigger.